src/engraving/dom/edit.cpp

```diff
@@ -1424,62 +1425,66 @@ void Score::cmdAddTimeSig(Measure* firstMeasure, staff_idx_t staffIdx, TimeSig*
                 timeSigOfCurrentStaffIndex->setDropTarget(false);
             }
         }
-    } else {
+    }
+    if (!measureDurationUnchanged) {
```

Comment:
I don't see much benefit in changes like this. To me, they make it only more confusing: with an `if`/`else`, it's clear that the two blocks are alternatives.

Extracting complex expressions into named variables, on the other hand, is alright; however, make sure that this does not lead to expressions being unnecessarily evaluated (for example when they are used in only one if/else branch).

Reply:
The problem for me with:
'''
} else { 
'''
is it's *such* a long way from the original condition that it's quite hard to tell exactly which 'if' statement it's associated with. I was scrolling up and trying to count brackets! 

I would use 'else' if I can see the 'if' but it's not even on the same screen really.

The code is also heavily nested and reducing the nesting feels good to me

Reply:
@brobrobrob Would it help to move the code inside the if-branch into its own (private) function? Based on the comment, it could be called something like `updateTimeSignatures`.  Then, the if/else branches aren't so far away from each other.

Reply:
Yes, that's a sensible suggestion.

There's a lot of work that needs doing to cmdAddTimeSig IMO (more DRY/extract it's own class), but I'm not experienced enough with C++ or the Musescore codebase to do it yet.

Reply:
I would propose undoing the change, and just leaving the `} else {` there. Long blocks are of course not ideal, but that is what almost all IDEs have folding for. 

The problem that I have with changing it into another `if` block, is that it obfuscates the semantics of the code. It is no longer clear that either one block or the other is executed, or that there is any relation between the blocks. 
When I see `if (b) {}` followed by `if (!b) {}` I would be quite confused: was the second one really supposed to be `if (!b)` and not `if (!c)`? Because otherwise it would surely have been `else`, right? Or is the point that `b` is modified inside the first `if` block? But where...?
To me, that kind of confusion is much worse than having to scroll up a little bit or click one time extra to fold some block of code.

A real solution could be indeed to move the method to its own file/class, and break it into smaller methods. I don't think it is very necessary to do that though. Also, while having long methods can be inconvenient, having a lot of small methods can be inconvenient too. I'm afraid it's to some extent inevitable that codebases of large overall size will also have more code per unit (i.e. per method, class, translation unit).

Reply:
@cbjeukendrup I completely disagree with everything you said, but it's your codebase not mine so have it the way you like it.

Reply:
Can we address the more pressing problem of:

This code doesn't actually work, but it did 2 years ago, why?